File: Sources/ClaudeUsageWidgetCore/Status/StatusAPI.swift
import Foundation

/// Reads Claude's public status page. No credentials: the endpoint is open.
public struct StatusAPI: Sendable {
    public static let endpoint = URL(string: "https://status.claude.com/api/v2/summary.json")!

    private let session: URLSession

    public init(session: URLSession = UsageAPI.connectivityAwareSession) {
        self.session = session
    }

    public func fetch() async throws -> ServiceStatus {
        var request = URLRequest(url: Self.endpoint)
        request.httpMethod = "GET"
        request.setValue("application/json", forHTTPHeaderField: "Accept")

        let data: Data
        let response: URLResponse
        do {
            (data, response) = try await session.data(for: request)
        } catch {
            throw UsageError.network(error.localizedDescription)
        }

        guard let http = response as? HTTPURLResponse else { throw UsageError.malformedResponse }
        try UsageAPI.validate(statusCode: http.statusCode)
        return try StatusDecoder.status(from: data)
    }
}

File: src/ClaudeUsageWidget.Core/Status/StatusApi.cs
using System.Net.Http.Headers;

namespace ClaudeUsageWidget.Core;

/// Reads Claude's public status page. No credentials: the endpoint is open.
public sealed class StatusApi
{
    public static readonly Uri Endpoint = new("https://status.claude.com/api/v2/summary.json");

    /// Shared instance so callers that don't inject their own HttpClient
    /// still reuse one connection pool instead of opening a fresh socket per
    /// refresh tick.
    private static readonly HttpClient SharedClient = new();

    private readonly HttpClient _client;

    public StatusApi(HttpClient? client = null)
    {
        _client = client ?? SharedClient;
    }

    public async Task<ServiceStatus> FetchAsync(CancellationToken ct = default)
    {
        using var request = new HttpRequestMessage(HttpMethod.Get, Endpoint);
        request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

        HttpResponseMessage response;
        try
        {
            response = await _client.SendAsync(request, ct);
        }
        catch (Exception ex) when (ex is not OperationCanceledException)
        {
            throw new UsageException(UsageError.Network(ex.Message));
        }

        using (response)
        {
            HttpValidation.Validate((int)response.StatusCode);
            var body = await response.Content.ReadAsStringAsync(ct);
            return StatusDecoder.Status(body);
        }
    }
}
